buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchDistribution.java

```diff
@@ -36,49 +34,6 @@ public String toString() {
         }
     }
 
-    public enum Type {
```

Comment:
We replace this Type enum with a proper interface. 
That way we can have proper public and internal Types and hide the Docker based distributions that are not supported by the public `TestClustersPlugin` and is only used in our internal builds 
Also switches with defaults are evil and we should avoid them IMO